src/types/CanonicalField.ts

```diff
@@ -0,0 +1,57 @@
+import { ajv } from '../ajv';
+import type { JSONSchemaType } from 'ajv';
+
+export interface CanonicalField {
+  id: number;
+  label: string;
+  shortCode: string;
+  fieldType: string;
+  createdAt: Date;
+}
+
+export const defaultCanonicalField = {
+  id: 0,
+  label: '',
+  shortCode: '',
+  fieldType: '',
+  createdAt: new Date(),
+};
+
+export const canonicalFieldSchema: JSONSchemaType<CanonicalField> = {
+  type: 'object',
+  properties: {
+    id: {
+      type: 'integer',
+    },
+    label: {
+      type: 'string',
+    },
+    shortCode: {
+      type: 'string',
+    },
+    fieldType: {
+      type: 'string',
+    },
+    createdAt: {
+      type: 'object',
+      required: [],
+      instanceof: 'Date',
+    },
+  },
+  required: [
+    'id',
```

Comment:
This will be required in read-only mode but not required when creating one. Then again we might be able to defer that to another endpoint that accepts POST, but I would like to think about how to best resolve the two worlds in advance.

Reply:
Yeah, I think in that sense it probably just shouldn't be listed as required in the "default" object.